[16.0][MIG] pos_customer_wallet

carmenbianca opened the pull request:
## Description



## Odoo task (if applicable)

https://gestion.coopiteasy.be/web#id=10799&action=475&active_id=492&model=project.task&view_type=form&menu_id=536

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

carmenbianca committed:
[ADD] pos_customer_wallet

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Display wallet balance on POS payment screen

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Display wallet balance on POS payment screen

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Make global module toggle available in pos.config

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Make global module toggle available in pos.config

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Journal demo data

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Journal demo data

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Mandate selected customer when using wallet payment method

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Mandate selected customer when using wallet payment method

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Check sufficient balance before POS purchase

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Check sufficient balance before POS purchase

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] account_customer_wallet: Write account_wallet to res.company instead of settings

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] account_customer_wallet: Write account_wallet to res.company instead of settings

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Recompute balance based on POS statement lines

This does not yet live-update in the POS. The window needs to be
reloaded to see the updated balance.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Recompute balance based on POS statement lines

This does not yet live-update in the POS. The window needs to be
reloaded to see the updated balance.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_account_wallet: Update wallet balance in POS

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_account_wallet: Update wallet balance in POS

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Update write_date on computation of wallet balance

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Update write_date on computation of wallet balance

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REM] account_customer_wallet: Remove some fields and api.depends lines

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REM] account_customer_wallet: Remove some fields and api.depends lines

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Tests

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet: Tests

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] pos_customer_wallet: Re-implement wallet computation using partner._invoice_total algorithm

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] pos_customer_wallet: Re-implement wallet computation using partner._invoice_total algorithm

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REM] account_customer_wallet: Custom wallet toggle; depend on whether account is set instead

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REM] account_customer_wallet: Custom wallet toggle; depend on whether account is set instead

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] account_customer_wallet: Search balance on all partners in family tree

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] account_customer_wallet: Search balance on all partners in family tree

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Avoid null error when no customer is selected

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Avoid null error when no customer is selected

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] account_customer_wallet: README.rst

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] account_customer_wallet: README.rst

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

Commit:
[UPD] Update pos_customer_wallet.pot

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

Commit:
[UPD] Update pos_customer_wallet.pot

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

remytms committed:
[ADD] pos_customer_wallet: Show balance on ticket

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

remytms committed:
[ADD] pos_customer_wallet: Show balance on ticket

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

remytms committed:
[FIX] pos_customer_wallet: typo

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

remytms committed:
[FIX] pos_customer_wallet: typo

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Partner form should not break for users without perms

Previously there was a permission error on trying to read the
`account.bank.statement.line` model.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Partner form should not break for users without perms

Previously there was a permission error on trying to read the
`account.bank.statement.line` model.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] *_customer_wallet: Create partner.get_all_partners_in_family

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] *_customer_wallet: Create partner.get_all_partners_in_family

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet_is_user

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] pos_customer_wallet_is_user

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] *_customer_wallet: Change log entries

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] *_customer_wallet: Change log entries

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
pos_customer_wallet 12.0.1.0.1

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
pos_customer_wallet 12.0.1.0.1

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] SCRLfs → SC

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] SCRLfs → SC

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

Commit:
[UPD] Update pos_customer_wallet.pot

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

Commit:
[UPD] Update pos_customer_wallet.pot

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
[UPD] changelog

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
[UPD] changelog

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
pos_customer_wallet 12.0.1.1.0

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
[UPD] README.rst

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
pos_customer_wallet 12.0.1.1.0

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
[UPD] README.rst

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] SCRL fs → SC

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] SCRL fs → SC

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[FIX] pos_customer_wallet : makes the modules working in an non proxy mode. use case : make a sell, pay in cash, do not select a customer. An error is raised, because order.pos.get_client() is null

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[FIX] pos_customer_wallet : makes the modules working in an non proxy mode. use case : make a sell, pay in cash, do not select a customer. An error is raised, because order.pos.get_client() is null

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] move overload function at the beginning of the javascript file, and add function comment

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] move overload function at the beginning of the javascript file, and add function comment

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[FIX] force user to select a customer if wallet payment are selected, even if the total amount is null

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[FIX] force user to select a customer if wallet payment are selected, even if the total amount is null

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[ADD] implement wallet credit with product in the Point of Sale front office

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[ADD] implement wallet credit with product in the Point of Sale front office

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : do not display wallet balance on ticket if no customer is selected

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : do not display wallet balance on ticket if no customer is selected

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : display wallet balance on client detail screen. (editable / non editable view)

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : display wallet balance on client detail screen. (editable / non editable view)

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] pos_customer_wallet : refactor code and create a sub function to compute wallet for bank statement lines

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] pos_customer_wallet : refactor code and create a sub function to compute wallet for bank statement lines

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : update partner wallet balance with pending paid pos order lines if product is 'is_customer_wallet_product'

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : update partner wallet balance with pending paid pos order lines if product is 'is_customer_wallet_product'

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] Add GRAP as coauthor of the wallet modules

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] Add GRAP as coauthor of the wallet modules

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[i18n] update pot files and add fr translations

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[i18n] update pot files and add fr translations

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet[_*]: Rename foodprint class

The class name was wrongly copied from some other place and does not
reflect what it's trying to represent.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet[_*]: Rename foodprint class

The class name was wrongly copied from some other place and does not
reflect what it's trying to represent.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : return early if no partners

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : return early if no partners

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[TR] update fr translation

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[TR] update fr translation

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Use where_calc method in pos_order_line computation

This should be nicer on multi-company databases.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Use where_calc method in pos_order_line computation

This should be nicer on multi-company databases.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>
Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[FIX] do not displayed 'undefined' where creating a new customer via UI

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[FIX] do not displayed 'undefined' where creating a new customer via UI

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[PERF] refactor get_wallet_balance_pos_order_line(). 6 requests for 1962.92ms -> 26 request 867.72 ms

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[PERF] refactor get_wallet_balance_pos_order_line(). 6 requests for 1962.92ms -> 26 request 867.72 ms

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] refactor _compute_customer_wallet_balance overload. Do not duplicate alogithm

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] refactor _compute_customer_wallet_balance overload. Do not duplicate alogithm

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : simplify PoS template

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : simplify PoS template

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : display futur balance, if the order is confirmed

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : display futur balance, if the order is confirmed

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : add possibility to have wallet negative

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[IMP] pos_customer_wallet : add possibility to have wallet negative

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[translation] *_customer_wallet : update translation

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[translation] *_customer_wallet : update translation

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] Move minimum_wallet_amount field from pos.config to account.journal, cortesy @carmenbianca

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] Move minimum_wallet_amount field from pos.config to account.journal, cortesy @carmenbianca

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] pos_customer_wallet : create a function is_balance_above_minimum in PoS

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[REF] pos_customer_wallet : create a function is_balance_above_minimum in PoS

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[FIX] allow user to access to the wallet amount even if they are not member of 'Point of Sale / User' group

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain committed:
[FIX] allow user to access to the wallet amount even if they are not member of 'Point of Sale / User' group

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

Commit:
[UPD] Update pos_customer_wallet.pot

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

Commit:
[UPD] Update pos_customer_wallet.pot

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
[UPD] README.rst

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
[UPD] README.rst

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet[_partner_is_user]: Pass required argument

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet[_partner_is_user]: Pass required argument

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
pos_customer_wallet 12.0.2.1.0

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
pos_customer_wallet 12.0.2.1.0

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet_[*]: Repair maintainer tags in manifest

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet_[*]: Repair maintainer tags in manifest

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[UPD] Run prettier

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[UPD] Run prettier

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
[UPD] README.rst

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
pos_customer_wallet 12.0.2.1.1

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
pos_customer_wallet 12.0.2.1.1

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

github-grap-bot committed:
[UPD] README.rst

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

Commit:
[UPD] Update pos_customer_wallet.pot

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

Commit:
[UPD] Update pos_customer_wallet.pot

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] pos_customer_wallet: black, isort, prettier

carmenbianca committed:
[IMP] pos_customer_wallet: black, isort, prettier

carmenbianca committed:
[MIG] pos_customer_wallet: Migration to 13.0

base_suspend_security's functionality is wrapped into Odoo's sudo by
default

carmenbianca committed:
[MIG] pos_customer_wallet: Migration to 13.0

base_suspend_security's functionality is wrapped into Odoo's sudo by
default

carmenbianca committed:
[MIG] pos_customer_wallet: Migration to 14.0

carmenbianca committed:
[MIG] pos_customer_wallet: Migration to 14.0

carmenbianca committed:
[MIG] pos_customer_wallet: Migration to 15.0

carmenbianca committed:
[MIG] pos_customer_wallet: Migration to 15.0

carmenbianca committed:
[MIG] pos_customer_wallet: Migration to 16.0

carmenbianca committed:
[MIG] pos_customer_wallet: Migration to 16.0

carmenbianca committed:
[REF] pos_customer_wallet: Use ORM instead of SQL queries

This change was also done in account_customer_wallet. It simplifies things.

This change is necessary for the 16.0 migration, but is not included in
its commit for clarity's sake.

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] pos_customer_wallet: Use ORM instead of SQL queries

This change was also done in account_customer_wallet. It simplifies things.

This change is necessary for the 16.0 migration, but is not included in
its commit for clarity's sake.

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REM] pos_customer_wallet: Remove bank statement line method of wallet computation

The pos session does not create these anymore. It creates pos.payment
records instead.

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REM] pos_customer_wallet: Remove bank statement line method of wallet computation

The pos session does not create these anymore. It creates pos.payment
records instead.

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Add some copyright headers

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Add some copyright headers

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca commented:
I think this is ported as far as I'm going to be able to port it for the moment.

Some notes:

- I removed base_suspend_security as a dependency. @legalsylvain I know that you introduced this dependency, and I think it had to do with multi-company support, but I cannot well remember why it was there. There should be a test for the use case.
- I did not test multi-company support.
- I did a decent amount of functional testing, but I doubtlessly missed some spots.
- I didn't test toggling off the `is_enabled_customer_wallet` button.
- I should probably write tests for the JavaScript…

legalsylvain commented on pos_customer_wallet/models/res_partner.py:
```diff
@@ -66,4 +89,5 @@ def get_wallet_balance_all(self, all_partner_ids, all_account_ids):
         res = super().get_wallet_balance_all(all_partner_ids, all_account_ids)
         res.append(self.get_wallet_balance_bank_statement_line(all_partner_ids))
```
if we add pos_payment, we should remove bank_statement_line ? or did i missed something ?

legalsylvain replied:
Forget it, I did'nt saw commit after this one.

legalsylvain commented:
Thanks a lot ! 

> I should probably write tests for the JavaScript…

;-)

> I removed base_suspend_security as a dependency. @legalsylvain I know that you introduced this dependency, and I think it had to do with multi-company support, but I cannot well remember why it was there. There should be a test for the use case.

let's start like that, and see if we face some issue.

carmenbianca commented on pos_customer_wallet/static/src/js/Screens/PaymentScreen/PaymentScreen.js:
```diff
@@ -0,0 +1,208 @@
+             * of the current customer, if defined.
+             */
+            async _finalizeValidation() {
+                var payment_wallet_amount =
+                    this.get_amount_debit_with_customer_wallet_journal()[0];
+                var product_wallet_amount =
+                    this.get_amount_credit_with_customer_wallet_product()[0];
+                var wallet_amount = payment_wallet_amount - product_wallet_amount;
+
+                var partner = this.currentOrder.get_partner();
+
+                if (partner) {
```
everything here should be wrapped in `if (partner)`

legalsylvain commented:
Is it ready to review ?

carmenbianca commented:
@legalsylvain yes, i think!

carmenbianca commented:
Problem found during functional testing: The balance is not reduced when making POS payments with the child of a partner. Needs further investigation.

carmenbianca commented:
Ignore previous comment. I was testing the wrong payment method ('Customer Account' instead of 'Customer Wallet')

I did find another bug, though. After closing the session, all changes to the customer wallet balance disappear, as if the POS session never happened.

carmenbianca commented:
Interesting. The account.move.lines generated by closing the POS do not contain partner_ids. This is going to be painful to research.

legalsylvain commented on pos_customer_wallet/static/src/js/Screens/PaymentScreen/PaymentScreen.js:
```diff
@@ -0,0 +1,206 @@
+
+    const WalletPaymentScreen = (PaymentScreen_) =>
+        class extends PaymentScreen_ {
+            /* eslint-disable no-unused-vars */
```
```suggestion

```

carmenbianca replied:
This is needed to satisfy eslint regarding the isForceValidate parameter. It's not smart enough to recognise that `await super.validateOrder(...arguments)` implies the use of isForceValidate.

legalsylvain commented on pos_customer_wallet/static/src/js/Screens/PaymentScreen/PaymentScreen.js:
```diff
@@ -0,0 +1,206 @@
+             * @param {Boolean} isForceValidate - Passed to super.
+             * @returns {Boolean} Whether the order is valid.
+             */
+            async validateOrder(isForceValidate) {
```
```suggestion
            async validateOrder() {
```

carmenbianca replied:
It doesn't seem right to me to adjust the function signature.

legalsylvain commented on pos_customer_wallet/static/src/js/Screens/PaymentScreen/PaymentScreen.js:
```diff
@@ -0,0 +1,206 @@
+             * @returns {Boolean} Whether the order is valid.
+             */
+            async validateOrder(isForceValidate) {
+                /* eslint-enable no-unused-vars */
```
```suggestion
```

legalsylvain commented on pos_customer_wallet/static/src/js/Screens/PaymentScreen/PaymentScreen.js:
```diff
@@ -0,0 +1,206 @@
+                            ),
+                        });
+                        return;
+                    }
```
useless in V16.
check ``payment_method.split_transactions`` will do the job.

legalsylvain commented:
Hi @carmenbianca. Thanks for porting this module.

Just did some tests.

Globally it works ! 

Regarding your problem. 
> Interesting. The account.move.lines generated by closing the POS do not contain partner_ids. This is going to be painful to research.

It is partially wrong. for the payment account move, if you check the checkbox split_transactions AND set the outstanding account : it will : 
- A) force user to select a customer. (so part of the javascript can be removed, see comment inline)
- B) set a partner in the account move. So it's OK. See : 

Ex : payment of 5€ with wallet : 

![image](https://github.com/coopiteasy/addons/assets/3407482/71b80a35-930c-44ad-b4df-9e16b051a8b8)

BUT : if you sale "wallet product", AFAIK, the account move does'nt contain the partner, if it is not possible to credit the wallet, saling wallet in the PoS. 

Ex : credit 999€, saling wallet product : 
![image](https://github.com/coopiteasy/addons/assets/3407482/2e09e681-976b-4d15-9675-309099b3c148)

Do you know how we could fix it ?

carmenbianca commented:
Thanks for the tip on `split_transactions` @legalsylvain 

> BUT : if you sale "wallet product", AFAIK, the account move does'nt contain the partner, if it is not possible to credit the wallet, saling wallet in the PoS.
>
> Ex : credit 999€, saling wallet product :
>
> Do you know how we could fix it ?

I've been looking at this for a while. I'm not sure what a good solution is. When closing the session, it correctly gets the income account from the wallet product for each pos.order.line that sells a wallet product. However, the amount for every such pos.order.line is eventually accumulated/aggregated into a single partner-less account.move.line on the session's account move.

There exists no `split_transactions` that we can use here, and the code does not seem to have the right hooks for us to force such a behaviour.

Furthermore, the session's account move is on the POS journal, not on the customer wallet journal. If we were to succeed in splitting the lines on the session's account move, we might still run afoul of some expectations made in account_customer_wallet. Trivially fixed, I think, but important to keep in mind.

Some solutions that I can think of, but haven't thought through:

1. Like above, try to force `split_transactions` behaviour on the session's account move. There are no good hooks for this, so we might have to do it retroactively. I'm not sure I like the idea of retroactively adjusting account moves.
2. Do some creative accounting. After the POS session is closed, create enough account moves to solve the problem, reading from the pos.order.lines.
3. Don't do anything. Or rather, include the pos.order.lines (with wallet products) when calculating the balance. We _already_ do some form of this for open pos sessions. Furthermore, the customer wallet is already credited the correct amount when closing the session.

I will expand later.

carmenbianca commented:
Now is later.

All three options proposed above have some flaws.

A full example. Find below the account.move of the POS session, where Alice bought 800€ of wallet product, and Bob bought 200€:

| Account                  | Partner | Debit | Credit |
|--------------------------|---------|-------|--------|
| Account Receivable (PoS) |         | 1000  |        |
| Customer Wallet          |         |       | 1000   |

# Option 1: Split on the pos session's account move

I've explained the flaws of option 1 fairly well. The account move is on the wrong journal, retroactively editing a closed account move seems like a bad idea, and there are no good hooks to make this happen.

But if we did make it happen, we would turn the example account move into this:

| Account                  | Partner | Debit | Credit |
|--------------------------|---------|-------|--------|
| Account Receivable (PoS) |         | 1000  |        |
| Customer Wallet          | Alice   |       | 800    |
| Customer Wallet          | Bob     |       | 200    |

# Option 2: Do creative accounting after the pos session closes

Option 2 is fine-ish, if a little verbose and difficult to retroactively comprehend. You have to undo the account.move.line against the customer wallet account and create new per-partner lines.

Given the example, we would create one or two new moves that look a little like this:

## Two moves against a temporary account

| Account         | Partner | Debit | Credit |
|-----------------|---------|-------|--------|
| Temporary       |         |       | 1000   |
| Customer Wallet |         | 1000  |        |

and

| Account         | Partner | Debit | Credit |
|-----------------|---------|-------|--------|
| Temporary       |         | 1000  |        |
| Customer Wallet | Alice   |       | 800    |
| Customer Wallet | Bob     |       | 200    |

## One move against the wallet account itself

Not sure if this is possible.

| Account         | Partner | Debit | Credit |
|-----------------|---------|-------|--------|
| Customer Wallet |         | 1000  |        |
| Customer Wallet | Alice   |       | 800    |
| Customer Wallet | Bob     |       | 200    |

# Option 3: Compute the balance from pos orders; don't touch the account moves any further

Option 3 seems fairly easy, but I'm not certain it's a good idea. One ramification is that you cannot uninstall `pos_customer_wallet` without data loss to the balances. This is not presently the case if all POS sessions are closed, I think.

But overall, I just feel like it would be (mis)using pos orders for the wrong purpose.

carmenbianca committed:
[IMP] pos_customer_wallet: Some user-facing documentation

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] pos_customer_wallet: Some user-facing documentation

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain commented:
Thanks a lot @carmenbianca for all the investigations ! 

> Option 3
Well. I don't think option 3 is OK : For the reason you mentionned, about the uninstallation, but also because the information, regarding the credit _should_ be in account.move.line. It's an important information. If some bridges are done with other accounting software, the data should be exported. (for exemple, in a software for accountant experts).

> The account move is on the wrong journal,

I'm not sure. for me saling wallet product should go in sale journal. Don't you think ? I'm pretty sure it is the current behaviour in v12.

Regarding option 2, I think writing a single extra account move is OK in a accounting point of view.

For the time being, I think option 1 has my preference, but I read the code that prepare the account move lines, and it is just horrible. ;-) So maybe option 2 will be mandatory.

What is your deadline for this developpement ? we could make a little code sprint during the oca days?

regards.

carmenbianca commented:
@legalsylvain We can definitely sprint on this at OCA Days!

I agree with you on option 3.

carmenbianca committed:
[IMP] pos_customer_wallet: Slightly refactor tests to use the same session

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] pos_customer_wallet: Slightly refactor tests to use the same session

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] pos_customer_wallet: After closing, correctly handle purchased wallet products

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] pos_customer_wallet: After closing, correctly handle purchased wallet products

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca commented:
@legalsylvain I wrote an implementation for this finally! It was tricky to get right, but it implements option 1.

It needs heaps more tests before this can be merged, but I thought I'd notify you.

carmenbianca committed:
[REF] pos_customer_wallet: Refactor to .esm.js

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] pos_customer_wallet: Refactor to .esm.js

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain commented:
I have to review this one ! thanks for the ping !

polchampion commented:
@carmenbianca à combien d'heures estimes-tu la finalisation du portage ?

carmenbianca committed:
[IMP] pos_customer_wallet: Add taxation test

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] pos_customer_wallet: Add taxation test

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca commented:
@legalsylvain i added one important test for taxation. this should technically never be needed, but it's good to know that everything is correctly implemented.

ping for review :)

edit: of course it breaks in CI but not locally...

legalsylvain commented:
> edit: of course it breaks in CI but not locally...

ahah ! let me know when it's green. Thanks !

legalsylvain commented:
@carmenbianca : I'm not sure. Are you waiting for my review, or Do I have to wait the CI is green ? 

thanks !

carmenbianca committed:
[FIX] pos_customer_wallet: Remove test

This assertion is different between the CI environment (4 elements) and
local tests (3 elements). This discrepancy was introduced after the
previous commit. Exactly why it is different, I do not know, but the
assertion is not important enough to waste hours debugging.

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] pos_customer_wallet: Remove test

This assertion is different between the CI environment (4 elements) and
local tests (3 elements). This discrepancy was introduced after the
previous commit. Exactly why it is different, I do not know, but the
assertion is not important enough to waste hours debugging.

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

legalsylvain commented on a:
little typo ?

carmenbianca replied:
i was re-running the CI :) you have an eagle eye.

i've simply removed a test assertion in the latest commit. commit message:

This assertion is different between the CI environment (4 elements) and
local tests (3 elements). This discrepancy was introduced after the
previous commit. Exactly why it is different, I do not know, but the
assertion is not important enough to waste hours debugging.

it is ready for review, as far as i am concerned.

legalsylvain commented on pos_customer_wallet/tests/test_balance.py:
```diff
@@ -68,8 +68,6 @@ def test_close_session(self):
         # One line for each partner
         self.assertEqual(len(move_lines), 2)
-        # Two credits and one debit on the move
-        self.assertEqual(len(move_lines[0].move_id.line_ids), 3)
```
could you just comment the line. So we can investigate in a second step ?

carmenbianca replied:
done

legalsylvain reviewed (commented):
thanks ! I'll take a look in the next week.

legalsylvain committed:
[IMP] account_customer_wallet: Avoid misconfiguration. No need to configure wallet account at product level, if the product is flagged as 'Customer Wallet Product'

carmenbianca commented:
hi @legalsylvain ! if you have time, this still needs some reviewing.

legalsylvain committed:
[IMP] account_customer_wallet: add a banner on payment form, to mention that it is possible to use Wallet Journal.
- Add same banner on payment wizard

legalsylvain commented:
> hi @legalsylvain ! if you have time, this still needs some reviewing.

Work in progress ! 

I just finished a functional review. All is OK until session is closed. At this time, the balance of the customer become wrong. Use case : 
- sell Wallet product to customer (by invoice) : Balance : 0 -> 1000€ : OK
- Open Point of Sale, sell to customer 100€ : Balance 1000 -> 900€ : OK
- Close Point of sale : Balance 900 -> 1000€ : KO.

What do you think ?

legalsylvain committed:
[FIX] portal_customer_wallet: remove useless dependency to website module

legalsylvain committed:
[DOC] account_customer_wallet: full update documentation, adding screenshots

legalsylvain committed:
[IMP] account_customer_wallet: new tree view to see all customer wallet amounts

legalsylvain committed:
[DOC] portal_customer_wallet: full update documentation, adding screenshots. + ROADMAP

legalsylvain committed:
[IMP] pos_customer_wallet : display wallet amount in partner list screen

legalsylvain committed:
[REF] pos_customer_wallet: use same name for the js file, as in Odoo point_of_sale module

legalsylvain committed:
[UPD] Update french translation

legalsylvain committed:
[REF] customer_wallet (renamed from account_customer_wallet)

legalsylvain committed:
[REF] customer_wallet_portal (renamed from portal_customer_wallet)

legalsylvain committed:
[REF] customer_wallet_pos (renamed from pos_customer_wallet)

legalsylvain committed:
[IMP] customer_wallet_pos: put minimum_wallet_amount storable

legalsylvain committed:
[FIX] customer_wallet_account: allow to use all views (form, kanban, etc...) when displaying partners with wallet account

legalsylvain committed:
[REF] customer_wallet_account: code simplification

legalsylvain committed:
[REF+IMP] customer_wallet*: Add possibility to view the history of the customer wallet balance

- Add a dedicated wizard and menu entry to see all the wallets.
- Refactor & Improve computation mechanism.
- Fix incorrect _customer_wallet_depends() if point_of_sale is installed
- Total refactor of the portal part. Display credit and debit + fix: display pos pending moves.

legalsylvain committed:
[IMP] customer_wallet: Add redistribute feature

legalsylvain committed:
[FIX] customer_wallet_account: initialize correctly new fields in the main compute function

legalsylvain committed:
[FIX] customer_wallet_pos: set it as autoinstallable, as otherwise, the core customer_wallet module doesn't work, when point of sale is installed

legalsylvain committed:
[META] update authors and maintainers

legalsylvain committed:
[DOC] customer_wallet: finish description and update translation

legalsylvain committed:
[DOC] customer_wallet_portal: finish description and update translation

legalsylvain committed:
[DOC] customer_wallet_pos: finish description and update translation

legalsylvain committed:
[FIX] customer_wallet: Do not write in _inverse function.

legalsylvain committed:
[IMP] customer_wallet: improve message

legalsylvain committed:
[IMP] customer_wallet*: always allow to close pos session, even if wallet minimum amount changed

legalsylvain committed:
[IMP] customer_wallet_pos: avoid credit and debit in a same order.

legalsylvain committed:
[FIX] do not raise an error, if use doesn't have right to write on res.company

legalsylvain committed:
[REF] customer_wallet : redistribution feature. do not use specific journal. Wallet Journal is enough

legalsylvain committed:
[REF] Update readme / index files

legalsylvain committed:
Update customer_wallet/__manifest__.py

Co-authored-by: hugues de keyzer <odoo@hugues.info>

legalsylvain committed:
Apply suggestions from @huguesdk

Co-authored-by: hugues de keyzer <odoo@hugues.info>

legalsylvain committed:
Apply suggestions from code review

Co-authored-by: hugues de keyzer <github@hugues.info>

legalsylvain committed:
[REF] account_wallet: simplify action in redistribute wizard

legalsylvain committed:
[REF] account_customer_waller: rename main.py file

legalsylvain committed:
[REF] customer_wallet_portal : Add Roadmap

legalsylvain committed:
[FIX] customer_wallet_portal: fix typo

legalsylvain committed:
[REF] customer_wallet: mention that accounting demo / test data are not optimal

legalsylvain committed:
Update customer_wallet/wizards/customer_wallet_redistribute_wizard.py

Co-authored-by: hugues de keyzer <odoo@hugues.info>

legalsylvain committed:
[FIX] customer_wallet: simplify redistribute feature

legalsylvain committed:
[FIX] customer_wallet: avoid to have the possibility to create new partners in wallet tree view

legalsylvain committed:
Update customer_wallet/wizards/customer_wallet_redistribute_wizard.py

Co-authored-by: hugues de keyzer <odoo@hugues.info>

legalsylvain committed:
[IMP] customer_wallet: format amount with currency

legalsylvain committed:
[REF] customer_wallet* : use of new odoo.command syntax

legalsylvain committed:
Update customer_wallet/wizards/customer_wallet_redistribute_wizard.py

Co-authored-by: hugues de keyzer <odoo@hugues.info>

legalsylvain committed:
[IMP] customer_wallet: add button on partner form view

legalsylvain committed:
[REF] customer_wallet: simplify code

legalsylvain committed:
[REF] customer_wallet_pos: remove useless eslint disabling

co-authored-by: hugues de keyzer <odoo@hugues.info>

legalsylvain committed:
[FIX] customer_wallet_pos: compare float

legalsylvain committed:
[REF] customer_wallet_pos: replace var by const / let

co-authored-by: hugues de keyzer <odoo@hugues.info>

legalsylvain committed:
[REF] update translation

huguesdk committed:
[IMP] *_customer_wallet: fix xml indentation

huguesdk committed:
[IMP] customer_wallet*: improve readme

huguesdk committed:
[IMP] customer_wallet: improve wizard buttons

* replace "save" and "discard" default buttons of the details wizard by
  a "close" button.
* improve styling of the buttons of the redistribute wizard.
* fix xml indentation.

huguesdk committed:
[IMP] customer_wallet_portal: improve layout

* fix alignment of amount and balance column values.
* use t-else instead of opposite test.
* replace t-esc by t-out.

huguesdk committed:
[IMP] customer_wallet: make cosmetic code changes

* rename model classes to match filename.
* remove empty lines between field definitions.

huguesdk committed:
[IMP] customer_wallet_pos: improve javascript code

* simplify javascript code.
* improve user messages.
* replace t-esc by t-out.

huguesdk committed:
[IMP] customer_wallet_pos: update roadmap

add known issue about sequence of operations to the roadmap.

huguesdk committed:
[IMP] customer_wallet_portal: improve translation

improve fr translation.

huguesdk committed:
[IMP] customer_wallet_pos: improve user messages

huguesdk committed:
[IMP] customer_wallet: correct user message

huguesdk committed:
customer_wallet_pos: simplify css

use a more sober and simple styling to display the customer wallet
amount on the pos payment screen.

huguesdk committed:
[IMP] customer_wallet: format datetime in wizard

format datetime as a date (since coming from account.move.line it has no
time) (taking the locale into account) in the customer wallet detail
wizard.

huguesdk committed:
[IMP] customer_wallet*: update contributors

huguesdk commented:
/ocabot merge nobump
